caddyhttp/proxy/reverseproxy.go

```diff
@@ -231,6 +232,26 @@ func (rp *ReverseProxy) UseInsecureTransport() {
 	}
 }
 
+func (rp *ReverseProxy) SetTransportCARoots(certPool *x509.CertPool) {
+	if rp.Transport == nil {
+		transport := &http.Transport{
+			Proxy:               http.ProxyFromEnvironment,
+			Dial:                defaultDialer.Dial,
+			TLSHandshakeTimeout: 10 * time.Second,
+			TLSClientConfig:     &tls.Config{RootCAs: certPool},
+		}
```

Comment:
Yes, something like that. Although I was thinking of a function that just returned an http.Transport that you could further customize before setting on the ReverseProxy.Transport struct field.

> the `rp.Transport.(*http.Transport)` is now always created, is that right?

Hmm, that does seem to be true now. And if it wasn't, I suppose the reverse proxy would basically break, because we depend on that concrete type so much.